tinygrad/shape/view.py

```diff
@@ -97,7 +98,7 @@ def reshape(self, new_shape: Tuple[sint, ...]) -> Optional[View]:
 
     # check if this is adding or removing 1s (only)
     # NOTE: this is optional, but removes most calls to (expensive!) merge_views (with mask, not optional)
-    if [x for x in self.shape if x != 1] == [x for x in new_shape if x != 1]:
+    if sum(1 for a,b in zip_longest([x for x in self.shape if x != 1],[x for x in new_shape if x != 1]) if a != b) <= 1:
```

Comment:
I don't understand this, what does this catch that the old one didn't?

Reply:
This catches reshapes like (3,4) --> (3, i) where an int gets replaced with a Variable and the other way round.

Reply:
Added a comment for this.

Reply:
oh I would handle this explicitly then

Reply:
updated it to (hopefully!) be more readable